test/fixtures/dialects/hive/array_types.sql

```diff
@@ -0,0 +1,14 @@
+-- simple
+select array[a, b, c] as arr
+from sch.tbl;
+
+-- bit harder
+select t.a
+from unnest(array[1, 3, 6, 12]) as t(f);
+
+-- complex
+select map_from_entries(array[
+    row('pending.freebet', pending_fb),
+    row('bonus.balance', bonus)
+])
+from sch.tbl;
```

Comment:
```suggestion
from sch.tbl;

select array['a', 'b', 'c'] as arr
from sch.tbl;

select array['a', null] as arr
from sch.tbl;
```

Reply:
Added in https://github.com/sqlfluff/sqlfluff/pull/2891/commits/d13db3defe9b03e51fa6e3bcd5ffbe143d92cfde

Reply:
skipping empty array coverage?

Reply:
Missed that one as aded after. Added now.